README.MD

```diff
@@ -5,75 +5,72 @@
 
-## Serialization
+## [Serialization](./serialization/)
 
-Abstract serialization library that supports any buffer implementation. Automatically (de)serializes
-classes marked with `@AutoSerializable`, which consist of fields of default types,
-`Proto4jSerializable` implementations or other `@AutoSerializable` instances.
+Abstract serialization library that supports any buffer implementation. Automatically (de-)serializes
+classes marked with [`@AutoSerializable`](./serialization/src/main/java/sexy/kostya/proto4j/serialization/annotation/AutoSerializable.java) consisting of fields of default types,
+[`Proto4jSerializable`](./serialization/src/main/java/sexy/kostya/proto4j/serialization/Proto4jSerializable.java) implementations or other [`@AutoSerializable`](./serialization/src/main/java/sexy/kostya/proto4j/serialization/annotation/AutoSerializable.java) members.
 
-Also supports inheritance: serializable class can extend other `@AutoSerializable`. In this case,
-all parent fields will also be serialized.
+Also supports inheritance: serializable class can extend other [`@AutoSerializable`](./serialization/src/main/java/sexy/kostya/proto4j/serialization/annotation/AutoSerializable.java) in which case
+all parent fields will also become transitively serialized.
 
-Whether you choose not to serialize some field, mark it with `@Transient`.
+Fields which should be ignored during serialization should be annotated with [`@Transient`](./serialization/src/main/java/sexy/kostya/proto4j/serialization/annotation/Transient.java).
 
-## Transport
+## [Transport](./transport/)
 
-Networking library with custom reliable UDP implementation.
+Networking library with custom UDP-based implementation with configurable reliability.
 
 ### Low level
 
-`Proto4jServer` and `Proto4jClient` allows you to transfer data using datagrams between sockets.
+[`Proto4jServer`](./transport/src/main/java/sexy/kostya/proto4j/transport/lowlevel/Proto4jServer.java) and [`Proto4jClient`](./transport/src/main/java/sexy/kostya/proto4j/transport/lowlevel/Proto4jClient.java) allow you to transfer data using datagrams between sockets.
 
-By default, any transmitted data is ordered, reliable, can be split into several UDP packets
-and combined back on receiver side, and is guaranteed to be delivered.
+By default, any transmitted data is ordered, reliable, is allowed to be split into several UDP packets
+and be combined back on receiver side and is guaranteed to be delivered.
 
 #### Packets structure
 
-All UDP packets that are being sent is of the following structure:
-- Header
-  - Packet length = 2 bytes
-  - Packet flags = 1 byte
-  - Sequence number = 4 bytes
-- Body = variable length
-- Signature = 4 bytes
+All UDP packets that are being sent are of the following structure:
+
+1. **Header**
+   1. **Length**: 2 bytes
+   2. **Flags**: 1 byte
+   3. **Sequence number**: 4 bytes
+2. **Body**: [*Length*] bytes
```

Comment:
Length that is specified in the header is the total length of the whole packet, not just the body's one.